mqtt/mqtt-edgehub/src/auth/authentication/edgehub.rs

```diff
@@ -28,30 +30,33 @@ impl EdgeHubAuthenticator {
         &self,
         context: AuthenticationContext,
     ) -> Result<Option<AuthId>, AuthenticateError> {
-        let req = EdgeHubAuthRequest::from_auth(&context);
+        let auth_req = EdgeHubAuthRequest::from_auth(&context);
+        let body = serde_json::to_string(&auth_req).map_err(AuthenticateError::SerializeRequest)?;
+        let req = Request::post(&self.url)
+            .body(Body::from(body))
+            .map_err(AuthenticateError::PrepareRequest)?;
 
-        let response = self
+        let http_res = self
             .client
-            .post(&self.url)
-            .json(&req)
-            .send()
+            .request(req)
             .await
             .map_err(AuthenticateError::SendRequest)?;
 
-        if response.status() != reqwest::StatusCode::OK {
-            return Err(AuthenticateError::ProcessResponse);
+        if http_res.status() != StatusCode::OK {
+            return Err(AuthenticateError::UnsuccessfullResponse(http_res.status()));
```

Comment:
nit: same question here, is it possible to get some more info from the body? @vipeller , does your authN api provides useful info in the body in case of an error?